File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0;
        int candidate = nums[0];
        for (int num:nums) {
            if (count == 0) {
                candidate = num;
                count += 1;
            } else if (num == candidate) {
                count ++;
            } else {
                count --;
            }
            System.out.print(count);
        }
        System.out.print(candidate);
        return candidate;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        res = nums[0]
        count = 0
        for num in nums:
            if num == res:
                count += 1
            elif num != res and count > 0:
                count -= 1
            elif num != res and count == 0:
                res = num
                count += 1
        return res
            